lte/gateway/python/magma/mobilityd/ip_allocator_factory.py

```diff
@@ -0,0 +1,75 @@
+import redis
+import logging
+from collections import defaultdict
+from typing import MutableMapping
+
+from magma.mobilityd.ip_descriptor import IPDesc
+from lte.protos.mconfig.mconfigs_pb2 import MobilityD
+
+from magma.mobilityd import mobility_store as store
+
+from .ip_allocator_dhcp import IPAllocatorDHCP
+from .ip_allocator_base import IPAllocator
+from .ip_allocator_static import IpAllocatorStatic
+from .ip_descriptor_map import IpDescriptorMap
+from .uplink_gw import UplinkGatewayInfo
+
+
+class IPAllocatorFactory:
```

Comment:
https://github.com/magma/magma/wiki/Contributing:-General-Code-Conventions#single-responsibilities

This component does too much because it contructs an IPAllocator instance but it also constructs the other state stores that are needed only by the IPAddressManager.

I don't think you actually need a factory class. Isolate any logic that needs to determine the backing store to main.py and instantiate all these dependency components there, passing them to the IPAddressManager via constructor injection

Reply:
Agreed, IPAllocator should be created inside IPAddressManager, and we could instantiate just the maps in main.

About injecting all the maps Into IPAddressManager. I think the constructor will have a lot of parameters, at least 5 maps plus 2 configs (mobilityd_conf + ip_allocator_type), are you ok  with that? Maybe the factory class could be useful to avoid inject too many parameters, what do you think?